provider/packet/environ.go

```diff
@@ -316,6 +366,114 @@ func (e *environ) StartInstance(ctx context.ProviderCallContext, args environs.S
 	}, nil
 }
 
+// supportedInstanceTypes returns the instance types supported by packet.
+func (e *environ) supportedInstanceTypes() ([]instances.InstanceType, error) {
+	plans, _, err := e.packetClient.Plans.List(new(packngo.ListOptions))
+	if err != nil {
+		return nil, errors.Annotate(err, "retrieving supported instance types")
+	}
+
+	var instTypes []instances.InstanceType
+
+nextPlan:
+	for _, plan := range plans {
+		if !validPlan(plan) {
+			continue
+		}
+
+		// TODO: Only work with this plan type for now to avoid
+		// spinning up more expensive instances while testing.
+		if plan.Name != "t1.small.x86" {
+			continue
+		}
+
+		var instArch string
+		switch {
+		case strings.HasSuffix(plan.Name, ".x86"):
+			instArch = arch.AMD64
+		case strings.HasSuffix(plan.Name, ".arm"):
+			instArch = arch.AMD64
+		default:
+			continue nextPlan
+
+		}
+
+		mem, err := parseMemValue(plan.Specs.Memory.Total)
+		if err != nil {
+			continue
+		}
+
+		instTypes = append(instTypes,
+			instances.InstanceType{
+				Id:       plan.ID,
+				Name:     plan.Name,
+				CpuCores: uint64(plan.Specs.Cpus[0].Count),
```

Comment:
I have noticed that `Specs.Cpus` is a slice but each entry has a CPU family and a count. Can you provide an example where plan spec contain multiple CPU entries? What would be the best way to map that into a CPU core count (just sum them together even if they are different CPU types?)?

Reply:
That seems reasonable to me. I'm not aware of a CPU type mix per plan or otherwise.

Reply:
I didn't find any examples of a plan having more than one entry for that slice.
I think it may be best to stick with the `[0].Count`.

I would need to confirm this, but without knowing this slice could be to offer alternative CPU sets for the same hardware model.  That would seem contrary to how `disks` are all included.

Reply:
> what would be the best way to map that into a CPU core count (just sum them together even if they are different CPU types?)

@detiber ran into a similar need for https://github.com/openshift/cluster-api-provider-equinix-metal/pull/1, the provider wants cores but the API only returns CPUs.

I'm going to raise an internal issue to see about getting this information added to the API, in the meanwhile, we will have to make do with CPU count.